crates/ruma-client-api/src/future/update_future.rs

```diff
@@ -11,39 +11,89 @@ pub mod unstable {
         api::{request, response, Metadata},
         metadata,
     };
+    use serde::{Deserialize, Serialize};
 
     const METADATA: Metadata = metadata! {
         method: POST,
         rate_limited: true,
-        authentication: None,
+        authentication: AccessToken,
         history: {
-            unstable => "/_matrix/client/unstable/org.matrix.msc4140/future/:token",
+            unstable => "/_matrix/client/unstable/org.matrix.msc4140/delayed_events/:future_id",
         }
     };
 
-    /// Request type for the [`update_future`](crate::future::update_future) endpoint.
+    /// The possible update actions we can do for updating a future.
+    #[derive(Serialize, Deserialize, Debug, Clone)]
+    #[serde(rename_all = "lowercase")]
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    pub enum UpdateAction {
```

Comment:
Ultimately, this should use the `StringEnum` derive macro to be able to handle unknown actions. It's fine if it's not fixed right now.